packages/core-plugin-api/src/analytics/Tracker.ts

```diff
@@ -21,6 +21,24 @@ import {
 } from '../apis';
 import { AnalyticsContextValue } from './';
 
+/**
+ * Global storing recent "gathered" mountpoint navigations.
+ */
+const gatheredNavigations: Array<{
+  action: string;
+  subject: string;
+  value?: number;
+  attributes?: AnalyticsEventAttributes;
+  context: AnalyticsContextValue;
+}> = [];
+
+/**
+ * Global storing recent routable extension renders.
+ */
+const routableExtensionRenders: Array<{
+  context: AnalyticsContextValue;
+}> = [];
```

Comment:
Since all we do is push and pop + clear, these can just be single optional values rather than arrays right? Feel that could potentially clean up the logic a bit too

Reply:
Do you think a global singleton from version bridge is warranted, @Rugvip?